fastlane: Better changelogs

rye opened the pull request:
This PR resolves #2822 by improving the iOS changelog format.

We now go from something like this:
```
*   e399b3ec Merge pull request #3045 from StoDevX/greenkeeper/react-native-0.57.0
|\  
| * 64d82112 remove fix-rn script
| * 79e66da2 disable [scrollEnabled] prop since it crashes RN@0.57.5
| *   40441dab Merge pull request #3185 from StoDevX/rn-0.57.5--remove-hsps
| |\  
| | * 5e916782 ios: Add HEADER_SEARCH_PATHS back to OneSignal targets
| | * 04d1da4f ios: Add back react-native-onesignal to pbxproj
| | * f0a42fd5 iOS: Try completely removing HEADER_SEARCH_PATHS
| * | 216d40d7 restore the accidentally-removed onesignal typedef
| * | c0003614 bump detox to v9.1.2
| * |   55a8ec62 Merge pull request #3184 from StoDevX/rn-0.57.5-remove-rc-from-hsp
| |\ \  
| | |/  
| | * 218edfd0 ios: Also remove ReactCommon from Release HSPs
| | * 9778d0c0 iOS: Try removing ReactCommon from the header search paths
| * | f4486778 remove react-navigation flow exclusion
| * | 7117a35e remove react-theme-provider flow exclusion
| * | ace369b7 update rn-linear-gradient to 2.4.4
| * | d19aef5e update callstack/react-theme-provider to 1.0.7
| * | 5872e53c RN 0.57.2 added [scrollEnabled] to the flow types
| * |   2e109eeb Merge pull request #3183 from StoDevX/rn-0.57.5--master-sync
| |\ \  
| | |/  
| |/|   
| | * f49bfd2b modules/lists/list-row: Remove unused eslint-disable directive
| | *   aea005fd Merge branch 'master' into greenkeeper/react-native-0.57.0
| | |\  
| |/ /  
| * | cf350ed1 yarn: Deduplicate the lockfile
| * | db893aa7 start update to v0.57.5
| * | 48978d03 bump eslint targeted flow version to 0.78
| * | 750d4c7c bump eslint target react version to 16.6
| * | 34e7a419 yarn: Upgrade react-native to 0.57.4
| * |   b672ab3f Merge branch 'master' into greenkeeper/react-native-0.57.0
| |\ \  
[...]
```
to something much more manageable and understandable like this:
```
Merged Pull Requests as of v2.6.3-470-ge399b3ec:

e399b3ec Merge pull request #3045 from StoDevX/greenkeeper/react-native-0.57.0
161270ad Merge pull request #3189 from StoDevX/express-bus-update
40441dab Merge pull request #3185 from StoDevX/rn-0.57.5--remove-hsps
55a8ec62 Merge pull request #3184 from StoDevX/rn-0.57.5-remove-rc-from-hsp
2e109eeb Merge pull request #3183 from StoDevX/rn-0.57.5--master-sync
7dbfc225 Merge pull request #3180 from StoDevX/jsc-android--236355.0.0
b3db5d0e Merge pull request #3179 from StoDevX/fastlane-refactor
647381c7 Merge pull request #3178 from StoDevX/danger-check-changelog
4f502fb9 Merge pull request #3177 from StoDevX/sync-changelog
46dc636a Merge pull request #3156 from StoDevX/sarn-website
a1aeefd8 Merge pull request #3176 from StoDevX/hours/pause-kitchen
8a22ebe1 Merge pull request #3174 from StoDevX/travis-xenial
d03bfd3f Merge pull request #3166 from StoDevX/remove-credentials-from-redux
d62c38ad Merge pull request #3172 from StoDevX/bundle-update
ec89f24d Merge pull request #3171 from StoDevX/circle-node-10
04be818d Merge pull request #3167 from StoDevX/disable-nav-persistence
6cb26732 Merge pull request #3108 from StoDevX/update-privacy
[...]
```

rye committed:
fastlane: Ungraph and use grep to only record merges

[GitHub] This commit resolves #2822.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>
Tested-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
fastlane: Update variable names in #git_changelog

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>
Tested-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
fastlane#git_changelog: Stop chomping lines

This doesn't actually do anything, especially on automated PR merges.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
fastlane#git_changelog: Add a header at the top

Using the longer form of `git describe --tags`, we get information
such as "how far ahead of master am I" and "what's the actual SHA
that's being built right now".

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

hawkrives reviewed (commented):
We could also be fancy and strip out the common starting text "Merged pull request" - that'd save 19 chars per PR

rye committed:
fastlane#git_changelog: Do some gsubbing

Note that gsub works with both Regexp's as arguments and
Strings---Strings are simply replaced verbatim with the substitution
text, whereas *all* occurrences of a Regexp's match group are
replaced.

Hence the name gsub.

Suggested-by: Hawken Rives <hawkrives@fastmail.fm>
Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
CHANGELOG: Add a note describing this PR's changes

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
fastlane#git_changelog: Don't print commit SHA

Even less verbosity! Yay.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
fastlane#git_changelog: Don't emdashify

Now, just replace "Merge pull request " with "" and "StoDevX/" with
the same.  This leaves "#ABCD: branch-name"

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

hawkrives commented on fastlane/lib/git.rb:
```diff
@@ -9,14 +9,13 @@ def git_changelog
+
+  "Merged Pull Requests as of #{describe}:\n\n#{pr_merges}"
+    .gsub(/Merge pull request |StoDevX\//,'')
+    .gsub(' from', ': ')
```
Won't the current version wind up with two spaces after the colon? I don't think we need to print the space after the colon.

```suggestion
    .gsub(' from', ':')
```

rye replied:
Why not replace ` from ` with `: `?  That's clearer to me.  I'll commit that.

rye replied:
Why did GitHub eat my spaces? We'll never know.  I meant to have a space both before and after `from`.

rye committed:
fastlane#git_changelog: Eat another space

This way everything is more balanced.

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>
